### ISSUE TITLE
lmh_incremental documentation

### ISSUE THREAD

**Issue by nilsbecker** [igarnier/prbnmcn-dagger#7]
the docs for `Lmh_incremental` read quite technical. i would say too technical for a newcomer.

it would be nice to state in a simple example the difference to `Lmh`. i looked at the Ising example and found a use of a shared array of spins. still i do not really get what is shared how and when; what happens in a single metropolis-hastings step and what would happen if doing the same naively using `Lmh`.

**Comment by igarnier** [comment]
I agree the API is a bit clunky, and the doc lacking. Ideally one would not need to have a dedicated type to express that some sub-graph is shared.
